src/components/Sections/services/formatServicesSections.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  IOCozyKonnector,
+  IOCozyTrigger,
+  IOCozyAccount
+} from 'cozy-client/types/types'
+
+import {
+  STATUS,
+  getAccountsFromTrigger,
+  getTriggersBySlug
+} from 'components/KonnectorHelpers'
+import { Section } from 'components/Sections/SectionsTypes'
+
+// List of categories that are allowed to be included even if pristine
+const categoriesWhitelist = [
+  'banking',
+  'clouds',
+  'isp',
+  'energy',
+  'public_service'
+]
+
+// Helper functions
+const isItemInMaintenance = (
+  maintenanceSlugs: Set<string>,
+  item: IOCozyKonnector
+): boolean => !maintenanceSlugs.has(item.slug)
+
+const getItemStatus = (accountsForKonnector: IOCozyAccount[]): number =>
+  accountsForKonnector && accountsForKonnector.length > 0
+    ? STATUS.OK
+    : STATUS.NO_ACCOUNT
+
+const sortItemsByStatusAndName = (
+  a: IOCozyKonnector & { status?: number },
+  b: IOCozyKonnector & { status?: number }
+): number => {
+  if (a.status === STATUS.OK && b.status !== STATUS.OK) return -1
+  if (a.status !== STATUS.OK && b.status === STATUS.OK) return 1
+  if (a.status === STATUS.NO_ACCOUNT && b.status !== STATUS.NO_ACCOUNT)
+    return -1
+  if (a.status !== STATUS.NO_ACCOUNT && b.status === STATUS.NO_ACCOUNT) return 1
+  return a.name.localeCompare(b.name)
+}
+
+const shouldIncludeSection = (section: Section, whitelist: string[]): boolean =>
+  whitelist.includes(section.name) || !section.pristine
+
+const sortSections = (
+  a: Section,
+  b: Section,
+  t: (key: string) => string
+): number => {
+  if (!a.pristine && b.pristine) return -1
+  if (a.pristine && !b.pristine) return 1
+  return t(`category.${a.name}`).localeCompare(t(`category.${b.name}`))
+}
+
+// New helper function for processing and sorting items within a category
+const processAndSortItems = (
+  items: IOCozyKonnector[],
+  allTriggers: Record<string, IOCozyTrigger>,
```

Comment:
This type shouldn't be `{ trigger: IOCozyTrigger }` as you cast it after with this type?

Reply:
types are all over the places because there are big differences between what is actual object in runtime and what the jsdoc is asking (in getTriggersByslug) etc. I'm going to try to clean this up for all files concerned

Reply:
updated